### PR TITLE
Добавлена обработка запросов

### PR DESCRIPTION
**Pull request by Rass00032**

### CODE REVIEW

**Comment by Rass00032** [comment]
Можете помочь, когда я запускаю приложение из IntelliJ IDEA и делаю запросы всё работает нормально, но кода создаю докер контейнер и запускаю его, post - запросы выполняются корректно, а get запросы выдают ошибку 1*.

Get: http://localhost:9090/stats?start=2022-09-03 1:00:22&end=2022-09-12 12:00:23&uris=/events/2&uris=/events/1&unique=true

если изменить формат даты на 2022-09-03T1:00:22
тогда ошибка 2*.


*ошибка 1: 
ERROR:  operator does not exist: timestamp without time zone >= character varying at character 89

java-explore-with-me-db-1  | 2023-08-21 16:20:21.433 UTC [67] HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.

java-explore-with-me-db-1  | 2023-08-21 16:20:21.433 UTC [67] STATEMENT:  SELECT app, uri, COUNT(DISTINCT IP) AS hits FROM request WHERE uri IN($1, $2) AND times BETWEEN $3 AND $4 GROUP BY uri 

stats-service              | 2023-08-21 16:20:21.440  WARN 1 --- [nio-9090-exec-3] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 0, SQLState: 42883

stats-service              | 2023-08-21 16:20:21.441 ERROR 1 --- [nio-9090-exec-3] o.h.engine.jdbc.spi.SqlExceptionHelper   : ERROR: operator does not exist: timestamp without time zone >= character varying

stats-service              |   Hint: No operator matches the given name and argument types. You might need to add explicit type casts.

stats-service              |   Position: 89

stats-service              | 2023-08-21 16:20:21.486 ERROR 1 --- [nio-9090-exec-3] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.springframework.dao.InvalidDataAccessResourceUsageException: could not extract ResultSet; SQL [n/a]; nested exception is org.hibernate.exception.SQLGrammarException: could not extract ResultSet] with root cause

stats-service              | 

stats-service              | org.postgresql.util.PSQLException: ERROR: operator does not exist: timestamp without time zone >= character varying

stats-service              |   Hint: No operator matches the given name and argument types. You might need to add explicit type casts.

stats-service              |   Position: 89

_______________________________________________________________________________________________________________________________________

*ошибка 2:
 2023-08-21 17:00:01.891 UTC [67] ERROR:  column "request.app" must appear in the GROUP BY clause or be used in an aggregate function at character 8

java-explore-with-me-db-1  | 2023-08-21 17:00:01.891 UTC [67] STATEMENT:  SELECT app, uri, COUNT(DISTINCT IP) AS hits FROM request WHERE uri IN($1, $2) AND times BETWEEN CAST($3 AS timestamp) AND CAST($4 AS timestamp) GROUP BY uri 

stats-server               | 2023-08-21 17:00:01.898  WARN 1 --- [nio-9090-exec-3] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 0, SQLState: 42803

stats-server               | 2023-08-21 17:00:01.898 ERROR 1 --- [nio-9090-exec-3] o.h.engine.jdbc.spi.SqlExceptionHelper   : ERROR: column "request.app" must appear in the GROUP BY clause or be used in an aggregate function

stats-server               |   Position: 8

stats-server               | 2023-08-21 17:00:01.911 ERROR 1 --- [nio-9090-exec-3] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.springframework.dao.InvalidDataAccessResourceUsageException: could not extract ResultSet; SQL [n/a]; nested exception is org.hibernate.exception.SQLGrammarException: could not extract ResultSet] with root cause
_______________________________________________________________________________________________________________________________________

**Comment by Rass00032** [comment]
Если использовать JPQL запросы выдаёт:
start 2022-09-03 11:00:22

stats-server               | end 2022-09-12 12:00:23

stats-server               | uri [/events/2, /events/1]

stats-server               | unique false

stats-server               | Hibernate: 

stats-server               |     SELECT

stats-server               |         app,

stats-server               |         uri,

stats-server               |         COUNT(*) AS hits 

stats-server               |     FROM

stats-server               |         request 

stats-server               |     WHERE

stats-server               |         uri IN(

stats-server               |             ?, ?

stats-server               |         ) 

stats-server               |         AND times BETWEEN CAST(? AS timestamp) AND CAST(? AS timestamp) 

stats-server               |     GROUP BY

stats-server               |         uri 

java-explore-with-me-db-1  | 2023-08-22 07:39:26.447 UTC [57] ERROR:  column "request.app" must appear in the GROUP BY clause or be used in an aggregate function at character 8

java-explore-with-me-db-1  | 2023-08-22 07:39:26.447 UTC [57] STATEMENT:  SELECT app, uri, COUNT(*) AS hits FROM request WHERE uri IN($1, $2) AND times BETWEEN CAST($3 AS timestamp) AND CAST($4 AS timestamp) GROUP BY uri 
_______________________________________________________________________________________________________________________________________

Не смотря на то что в коде у  "GROUP BY r.uri,r.app " есть параметр "r.app"

Запрос:

 @Query("SELECT new ru.practicum.dto.StatsDTO(r.app, r.uri, COUNT(*)) " +
            "FROM RequestModel r " +
            "WHERE r.uri IN(:uris) AND r.timestamp BETWEEN CAST(:start AS timestamp) AND CAST(:end AS timestamp) " + // /events/2,'/events/1' '2022-09-03 11:00:22' '2022-09-12 12:00:23'
            "GROUP BY r.uri,r.app;")
    List<StatsDTO> statisticsAllIp(String start, String end, List<String> uris);
_______________________________________________________________________________________________________________________________________

File: docker-compose.yml
Comment (stfaustoff) [inline docker-compose.yml:1]: Из файла не надо было убирать main сервис и его БД, так как в дальнейшем он понадобится.
Лучше сейчас все настроить чтобы потом на это время не терять

File: stats-service/src/main/java/ru/practicum/controller/StatsController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import ru.practicum.service.StatsServiceImpl;
+
+import java.util.List;
+
+@RestController
+public class StatsController {
+    @Autowired
+    private StatsServiceImpl service;
+    @Autowired
+    private ModelMapper mapper;
```
Comment (stfaustoff) [inline stats-service/src/main/java/ru/practicum/controller/StatsController.java:18]: Кажется что это поле здесь лишнее

File: stats-service/src/main/java/ru/practicum/controller/StatsController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public RequestModel creatingStats(@RequestBody RequestDto request) {
+        return service.save(request);
+    }
+
+    @GetMapping("/stats")
+    public List<StatsDTO> getStats(@RequestParam String start,
+                                   @RequestParam String end,
+                                   @RequestParam List<String> uris,
+                                   @RequestParam boolean unique) {
+        System.out.println("start " + start);
```
Comment (stfaustoff) [inline stats-service/src/main/java/ru/practicum/controller/StatsController.java:31]: Данные в консоль лучше всего выводить через log

File: stats-service/src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+spring.sql.init.mode=always
+spring.jpa.show-sql=true
+
+logging.level.ru.practicum.shareit=DEBUG
+logging.level.org.springframework.orm.jpa=INFO
+logging.level.org.springframework.transaction=INFO
+logging.level.org.springframework.transaction.interceptor=TRACE
+logging.level.org.springframework.orm.jpa.JpaTransactionManager=DEBUG
+
+#spring.jpa.hibernate.ddl-auto=create
```
Comment (stfaustoff) [inline stats-service/src/main/resources/application.properties:15]: Комментарии надо почистить во всех местах где они есть

**Review by stfaustoff — COMMENTED** [review]
Относительно запросов которые не выполняются - я бы предложил пересмотреть сам запросов, возможно стоит отказаться от кастов и between, заменив их на что-нибудь попроще
А так же обрати внимание что не все проверки гита прошли.

File: docker-compose.yml
Comment (stfaustoff) [reply docker-compose.yml:1]: Так же обрати внимание на то что требуется реализовать клиент и dto библиотеку

**Review by stfaustoff — APPROVED** [review]

**Pull request merged**